backend/dataall/modules/dataset_sharing/aws/kms_client.py

```diff
@@ -1,10 +1,38 @@
 import logging
+import json
 
 from dataall.base.aws.sts import SessionHelper
 from botocore.exceptions import ClientError
 
 log = logging.getLogger(__name__)
 
+DATAALL_BUCKET_KMS_DECRYPT_SID = 'DataAll-Bucket-KMS-Decrypt'
+DATAALL_KMS_PIVOT_ROLE_PERMISSIONS_SID = 'KMSPivotRolePermissions'
+DATAALL_ACCESS_POINT_KMS_DECRYPT_SID = 'DataAll-Access-Point-KMS-Decrypt'
+
```

Comment:
I thought about it. 
BUT
1. This method is only used here.
2. We call create_policy in many places, so, it's easy to forget to call _remove_malformed_principal before it. Here, on the other hand, it makes sense: if we have ClientError, we try to fix it.

I don't insist, but I think, it's more elegant to leave it here.

Reply:
Let me have a look at how the code looks after changes because it might be good now. I would be strict with the definition of layers, because we can end up mixing responsibilities and having a code that are just exceptions to the rule

Reply:
So, we are calling `put_key_policy` and that's when in case of MalformedPolicy we remove the malformed principals. We are doing all the policy manipulation in the s3_X_share_managers except for this piece that is a bit hidden in the aws layer. I propose to go for:
- kms client = put_key_policy that just calls boto3 KMS put_key_policy and rasies errors

There are 4 places (+tests) where we call that function, we could insert there the logic of put_policy -> raise exception -> remove malformed -> try again to put_policy. e.g. `dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py:723`

```
                try:
                    kms_client.put_key_policy(kms_key_id, json.dumps(existing_policy))
                except 'MalformedPolicyDocumentException': #catch error better
                    clean_policy = self._remove_malformed_principals(existing_policy)
                    kms_client.put_key_policy(kms_key_id, json.dumps(clean_policy))
```